File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/27_remove-element/solutions/cpp/solution.cpp
#include <vector>

class Solution {
public:
    int removeElement(std::vector<int>& nums, int val) {
        int insertPos = 0;

        for (int num : nums) {
            if (num != val) {
                nums[insertPos] = num;
                insertPos++;
            }
        }

        return insertPos;
    }
};

File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/27_remove-element/solutions/python3/solution.py
class Solution:

    def removeElement(self, nums: list[int], val: int) -> int:
        k = 0  # Start k at 0 to handle the empty or no valid case properly

        for num in nums:
            if num != val:
                nums[k] = num
                k += 1

        return k
